sdcm/sla/libs/sla_utils.py

```diff
@@ -0,0 +1,241 @@
+            return threads
+
+        def cache_only(max_rows_for_read):  # pylint: disable=unused-variable
+            if not max_rows_for_read:
+                max_rows_for_read = int(num_of_partitions * 0.3)
+            return 'seq=1..%d' % max_rows_for_read
+
+        # Read from cache and disk
+        def mixed(max_rows_for_read):  # pylint: disable=unused-variable
+            if not max_rows_for_read:
+                max_rows_for_read = num_of_partitions
+            return "'dist=gauss(1..%d, %d, %d)'" % (max_rows_for_read,
+                                                    int(max_rows_for_read / 2),
+                                                    int(max_rows_for_read * 0.05))
+
+        def disk_only(max_rows_for_read):  # pylint: disable=unused-variable
+            if not max_rows_for_read:
+                max_rows_for_read = int(num_of_partitions * 0.3)
+            return 'seq=%d..%d' % (max_rows_for_read, max_rows_for_read + int(num_of_partitions * 0.25))
+
+        rate = locals()[c_s_workload_type]()  # define -rate for c-s command depend on workload type
+        pop = locals()[load_type](max_rows_for_read)  # define -pop for c-s command depend on load type
+
+        params = {'n': num_of_partitions, 'user': role.name, 'password': role.password, 'pop': pop,
+                  'duration': '%dm' % stress_duration_min, 'threads': rate}
+        if kwargs:
+            params.update(kwargs['kwargs'])
+        c_s_cmd = stress_command.format(**params)
+        LOGGER.info("Created cassandra-stress command: %s", c_s_cmd)
+
+        return c_s_cmd
+
+    @staticmethod
+    def one_run_c_s_stats(read_run, user_name, statistic_name, tester):
+        res = tester.get_stress_results(queue=read_run, store_results=False)
+        stat_rate, username = None, None
+        if res:
+            stat_rate = res[0].get(statistic_name)
+            username = res[0].get('username')
+
+        if not (stat_rate and username):
+            LOGGER.error("Stress statistics are not received for user %s. Can't complete the test", user_name)
+            return None
+
+        return stat_rate, username
+
+    def get_c_s_stats(self, read_queue, users, statistic_name, tester):
+        role_names = [user['role'].name for user in users]
+
+        results = {}
+        for i, read in enumerate(read_queue):
+            stat_rate, username = self.one_run_c_s_stats(read_run=read, user_name=role_names[i],
+                                                         statistic_name=statistic_name, tester=tester)
+
+            if stat_rate is None:
+                return stat_rate
+
+            assert username == role_names[i], \
+                f'Expected that stress was run with user "{role_names[i]}" but it was "{username}"'
+
+            results[username] = float(stat_rate)
+
+        return results
+
+    @staticmethod
+    def validate_if_scylla_load_high_enough(start_time, wait_cpu_utilization, prometheus_stats):
```

Comment:
this seems like a not just SLA related helper function

Reply:
moved to `utils/common.py`